csharp/source/autowrap/csharp/csharp.d

```diff
@@ -0,0 +1,1104 @@
+    rangeDef.sliceEnd ~= "            if (typeof(T) == typeof(bool)) return new Range<T>(RangeFunctions.Bool_Slice(_slice, new IntPtr(begin), _slice.length), DStringType.None);" ~ newline;
+    rangeDef.sliceRange ~= "            if (typeof(T) == typeof(bool)) return new Range<T>(RangeFunctions.Bool_Slice(_slice, new IntPtr(begin), new IntPtr(end)), DStringType.None);" ~ newline;
+    rangeDef.appendValues ~= "            if (typeof(T) == typeof(bool)) { this._slice = RangeFunctions.Bool_AppendValue(this._slice, (bool)(object)value); }" ~ newline;
+    rangeDef.appendArrays ~= "            if (typeof(T) == typeof(bool)) { range._slice = RangeFunctions.Bool_AppendSlice(range._slice, source._slice); return range; }" ~ newline;
+
+    rangeDef.enumerators ~= "                else if (typeof(T) == typeof(string) && _strings != null && _type == DStringType.None) yield return (T)(object)_strings[(int)i];" ~ newline;
+    rangeDef.getters ~= "                else if (typeof(T) == typeof(string) && _strings != null && _type == DStringType.None) return (T)(object)_strings[(int)i];" ~ newline;
+    rangeDef.setters ~= "                else if (typeof(T) == typeof(string) && _strings != null && _type == DStringType.None) _strings[(int)i] = (string)(object)value;" ~ newline;
+    rangeDef.sliceEnd ~= "            else if (typeof(T) == typeof(string) && _strings != null && _type == DStringType.None) return new Range<T>((IEnumerable<T>)_strings.Skip((int)begin));" ~ newline;
+    rangeDef.sliceRange ~= "            else if (typeof(T) == typeof(string) && _strings != null && _type == DStringType.None) return new Range<T>((IEnumerable<T>)_strings.Skip((int)begin).Take((int)end - (int)begin));" ~ newline;
+    rangeDef.appendValues ~= "            else if (typeof(T) == typeof(string) && this._strings != null && this._type == DStringType.None) { this._strings.Add((string)(object)value); }" ~ newline;
+    rangeDef.appendArrays ~= "            else if (typeof(T) == typeof(string) && range._strings != null && range._type == DStringType.None) { foreach(T s in source) range._strings.Add((string)(object)s); return range; }" ~ newline;
+
+    generateStringBoilerplate("string", "String");
+    generateStringBoilerplate("wstring", "Wstring");
+    generateStringBoilerplate("dstring", "Dstring");
+
+    generateRangeDef!byte(libraryName);
+    generateRangeDef!ubyte(libraryName);
+    generateRangeDef!short(libraryName);
+    generateRangeDef!ushort(libraryName);
+    generateRangeDef!int(libraryName);
+    generateRangeDef!uint(libraryName);
+    generateRangeDef!long(libraryName);
+    generateRangeDef!ulong(libraryName);
+    generateRangeDef!float(libraryName);
+    generateRangeDef!double(libraryName);
+}
+
+private string writeCSharpBoilerplate(string libraryName, string rootNamespace) {
+    string returnTypesStr = string.init;
+    foreach(string rt; returnTypes.byValue) {
+        returnTypesStr ~= rt;
+    }
+
+    return "namespace Autowrap {
```

Comment:
With a string this large in foreign syntax, I'd use a string import and have it in a .cs file.